src/cluster/kv/etcd/store_test.go

```diff
@@ -125,6 +126,28 @@ func TestNoCache(t *testing.T) {
 	require.Equal(t, 0, len(store.(*client).cacheUpdatedCh))
 }
 
+func TestCacheDirCreation(t *testing.T) {
+	ec, opts, closeFn := testStore(t)
+
+	tdir, err := ioutil.TempDir("", "m3tests")
+	defer os.RemoveAll(tdir)
+	require.NoError(t, err)
+
+	cdir := path.Join(tdir, randSeq(10))
+	fname := path.Join(cdir, "mk3kv.json")
+	opts = opts.SetCacheFileFn(func(string) string {
```

Comment:
Instead of this, can you test to make sure the `dir` you created is compatible with `c.opts.Prefix()`?, try some writes and then read from the cache maybe.

Reply:
Done